### PR TITLE
Update environment-variables.mdx

### LINKED ISSUE #1871: Migrating from TF_ENCRYPTION environment variable to Early Variable Evaluation not working
### OpenTofu Version

```shell
OpenTofu v1.8.0
on windows_amd64
+ provider registry.opentofu.org/hashicorp/google v4.51.0
+ provider registry.opentofu.org/hashicorp/local v2.5.1
+ provider registry.opentofu.org/hashicorp/time v0.12.0
```

### OpenTofu Configuration Files

`terraform.tfvars`:
```hcl
passphrase = "correct horse battery staple"
```

`variables.tofu`:
```hcl
variable "passphrase" {
  description = "State encryption passphrase"
  type = string
}
```

`terraform.tofu`:
```hcl
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.51.0"
    }
  }

  encryption {
    key_provider "pbkdf2" "my_key" {
      passphrase = var.passphrase
    }

    method "aes_gcm" "state" {
      keys = key_provider.pbkdf2.my_key
    }

    state {
      method = method.aes_gcm.state
      enforced = true
    }
  }
}
```

### Debug Output

This is not a gist, but the output is fairly short:

```
2024-07-29T14:44:03.146-0400 [INFO]  OpenTofu version: 1.8.0
2024-07-29T14:44:03.147-0400 [DEBUG] using github.com/hashicorp/go-tfe v1.36.0
2024-07-29T14:44:03.147-0400 [DEBUG] using github.com/opentofu/hcl/v2 v2.0.0-20240416130056-03228b26f391
2024-07-29T14:44:03.147-0400 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2024-07-29T14:44:03.147-0400 [DEBUG] using github.com/zclconf/go-cty v1.14.4
2024-07-29T14:44:03.147-0400 [INFO]  Go runtime version: go1.21.3
2024-07-29T14:44:03.147-0400 [INFO]  CLI args: []string{"C:\\Users\\jpitlor\\AppData\\Local\\Programs\\OpenTofu\\tofu.exe", "init"}
2024-07-29T14:44:03.148-0400 [TRACE] Stdout is a terminal of width 188
2024-07-29T14:44:03.148-0400 [TRACE] Stderr is a terminal of width 188
2024-07-29T14:44:03.148-0400 [TRACE] Stdin is a terminal
2024-07-29T14:44:03.151-0400 [DEBUG] Attempting to open CLI config file: C:\Users\jpitlor\AppData\Roaming\terraform.rc
2024-07-29T14:44:03.152-0400 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-07-29T14:44:03.152-0400 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-07-29T14:44:03.152-0400 [DEBUG] ignoring non-existing provider search directory C:\Users\jpitlor\AppData\Roaming\terraform.d\plugins
2024-07-29T14:44:03.153-0400 [DEBUG] ignoring non-existing provider search directory C:\Users\jpitlor\AppData\Roaming\HashiCorp\Terraform\plugins
2024-07-29T14:44:03.153-0400 [DEBUG] Found the config directory: C:\Users\jpitlor\AppData\Roaming\terraform.d
2024-07-29T14:44:03.153-0400 [INFO]  CLI command args: []string{"init"}

Initializing the backend...
2024-07-29T14:44:03.564-0400 [TRACE] Meta.Backend: no config given or present on disk, so returning nil config
2024-07-29T14:44:03.564-0400 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2024-07-29T14:44:03.564-0400 [DEBUG] New state was assigned lineage "fd57940f-efa1-775f-0a92-ad09597dd48e"
2024-07-29T14:44:03.564-0400 [TRACE] Meta.Backend: using default local state only (no backend configuration, and no existing initialized backend)
2024-07-29T14:44:03.564-0400 [TRACE] Meta.Backend: instantiated backend of type <nil>
2024-07-29T14:44:03.564-0400 [DEBUG] checking for provisioner in "."
2024-07-29T14:44:03.564-0400 [DEBUG] checking for provisioner in "C:\\Users\\jpitlor\\AppData\\Local\\Programs\\OpenTofu"
2024-07-29T14:44:03.565-0400 [TRACE] Meta.Backend: backend <nil> does not support operations, so wrapping it in a local backend
2024-07-29T14:44:03.565-0400 [TRACE] backend/local: state manager for workspace "default" will:
 - read initial snapshot from terraform.tfstate
 - write new snapshots to terraform.tfstate
 - create any backup at terraform.tfstate.backup
2024-07-29T14:44:03.565-0400 [TRACE] statemgr.Filesystem: reading initial snapshot from terraform.tfstate
Error refreshing state: decryption failed for all provided methods: attempted decryption failed for state: decryption failed: cipher: message authentication failed
```

### Expected Behavior

The state should be decrypted

### Actual Behavior

The state is not decrypted

### Steps to Reproduce

1. `tofu init`

### Additional Context

The state file was previously created in OpenTofu 1.7 where, before executing `tofu init`, I would execute this:

```
export TF_ENCRYPTION=$(cat <<EOF
key_provider "pbkdf2" "my_key" {
  passphrase = "correct horse battery staple"
}
EOF
)
```

I am also seeing this issue on Windows, M2 Mac, and WSL.

And when I do that today, in 1.8.0, everything still works as it did before! So I just want to know if there's some steps I should be taking to migrate between the two strategies. Any weird quirks of either implementation that are relevant?

### References

_No response_

### PR DESCRIPTION
**Pull request by jpitlor**
Add warning that dollar sign in encryption key secret environment variables may get mangled by shells

Resolves #1871

## Target Release

1.9.0

## Checklist

- [x] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [x] I have not used an AI coding assistant to create this PR.
- [x] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.

### Go checklist

 

- [ ] I have run golangci-lint on my change and receive no errors relevant to my code.
- [ ] I have run existing tests to ensure my code doesn't break anything.
- [ ] I have added tests for all relevant use cases of my code, and those tests are passing.
- [ ] I have only exported functions, variables and structs that should be used from other packages.
- [ ] I have added meaningful comments to all exported functions, variables, and structs.

### Website/documentation checklist

- [x] I have locally started the website as [described here](https://github.com/opentofu/opentofu/blob/main/website/README.md) and checked my changes.

I attempted to check the changes with `docker-compose up --build`, but for some reason any changes I made weren't showing up. Is there a step I probably forgot to read?

Also, I'm assuming this is the wrong branch to be targeting, but it's what Github suggested. What should I be targeting instead? `main`?

### CODE REVIEW

File: website/docs/cli/config/environment-variables.mdx [hunk 1]
```diff
@@ -196,3 +196,7 @@ The `TF_ENCRYPTION` environment variable is an alternate method of specifying th
 # Add/Override encryption key_provider.static.mykp
 export TF_ENCRYPTION='key_provider "static" "mykp" { key = "6f6f706830656f67686f6834616872756f3751756165686565796f6f72653169" }'
 ```
+
+:::warning
+If the key (or similar) has a dollar sign in it, beware that your shell may interpret that as variable interpolation. Make sure the environment variable value is enclosed in single quotes, or escape it with a backslash to prevent your secret from becoming changed by your shell without you realizing.
```
[the remark this answers is not in the document]
Comment (Evi1Pumpkin) [reply website/docs/cli/config/environment-variables.mdx:201]: Maybe we can make the warning more general? If this is a potential issue for users and difficult to understand, I think it's a good idea to provide a warning.

[the remark this answers is not in the document]
Comment (jpitlor) [reply website/docs/cli/config/environment-variables.mdx:201]: I would totally be open to making the warning more generic! 

Honestly, even if the verdict is that this is out of scope, I would be happy knowing #1871 would probably show up in a Google search. And in hindsight, I probably should've known not to put a dollar sign in a bash string, but it totally slipped my mind as I told Bitwarden to generate something random.

How does this sound?

> If the key (or similar) has non-alphanumeric characters in it, beware that your shell may not interpret them literally. Special characters are shell dependent, but some common examples are dollar signs for variable interpolation or backslashes for character escaping.
>
> Make sure your secret doesn't get changed by your shell without you realizing. This is also shell dependent, but common ways of avoiding this are using single quotes or escaping special characters with a backslash.

[the remark this answers is not in the document]
Comment (cam72cam) [reply website/docs/cli/config/environment-variables.mdx:201]: That sounds good, I like the more generic approach here :+1:

**Comment by jpitlor** [comment]
Is there anyone who can answer either of these questions? I just want to make sure I do all I can to make the review process easy for you all!

> I attempted to check the changes with docker-compose up --build, but for some reason any changes I made weren't showing up. Is there a step I probably forgot to read?

> Also, I'm assuming this is the wrong branch to be targeting, but it's what Github suggested. What should I be targeting instead? main?

**Comment by cam72cam** [comment]
Target main, we will then take care of backporting from there.

**Comment by jpitlor** [comment]
Thanks! I figured v1.8 was probably wrong, but it was the default GitHub picked for me, so I figured we could fix it later.

Ahhh, that makes sense. Thanks for the pointer! I can confirm my changes worked, so I checked that box off. Not sure why all the tests aren't reporting back though...

**Review by Evi1Pumpkin — APPROVED** [review]
Thank you for improving our docs! 🙇‍♀️

**Pull request merged**